### ISSUE TITLE
Add a small check to a tensor operator= if it is overriding a externally managed tensor 

### ISSUE THREAD

**Issue by zhoonit** [nnstreamer/nntrainer#874]
As some tensors' memory are managed outside of it's scope, it would be great if we can warn users if she is overriding a managed memory.

One boolean + if inside tensor operator= would do the job.

**Comment by kparichay** [comment]
can you give an example of this?

**Comment by zhoonit** [comment]
A tensor can hold a external memory using `Tensor::Map(float *buf, const TensorDim &d, int offset)`

```cpp
// float * data;
Tensor A = Tensor::Map(data, dim, 0);
A = B.multiply(C); 
```

In this case, A is no longer holding original memory but overriden.
This might be intended overriding but if it is not, it is hard to track this issue.
We could warn a user if the underlying data is not managed with others and but this will be overriden at the copy constructor. Maybe with `DEBUG` or `CHECK_TENSOR_MEMORY` definition enabled

**Comment by kparichay** [comment]
Yeah, let's do it under `DEBUG`, but just remember, this might give a lot of warnings.
